akka-http-marshallers-scala/akka-http-spray-json/src/main/scala/akka/http/scaladsl/marshallers/sprayjson/SprayJsonSupport.scala

```diff
@@ -23,23 +23,25 @@ import scala.language.implicitConversions
 trait SprayJsonSupport {
   implicit def sprayJsonUnmarshallerConverter[T](reader: RootJsonReader[T]): FromEntityUnmarshaller[T] =
     sprayJsonUnmarshaller(reader)
+
   implicit def sprayJsonUnmarshaller[T](implicit reader: RootJsonReader[T]): FromEntityUnmarshaller[T] =
     sprayJsValueUnmarshaller.map(jsonReader[T].read)
-  implicit def sprayJsonByteStringUnmarshaller[T](implicit reader: RootJsonReader[T]): FromByteStringUnmarshaller[T] =
+  implicit def sprayJsValueUnmarshaller: FromEntityUnmarshaller[JsValue] =
+    Unmarshaller.byteStringUnmarshaller
+      .forContentTypes(`application/json`)
```

Comment:
According to http://www.iana.org/assignments/media-types/application/json adding charset parameter should have no effect:

```
Note:  No "charset" parameter is defined for this registration.
   Adding one really has no effect on compliant recipients.
```

So, yes, we might introduce a regression for people relying on non-standard behavior. In reality, it might not be a big deal because almost everyone seems to be on UTF-8 nowadays. It is still possible to create a custom unmarshaller if you need to deal with non-standard data. If it turns out that this is a pain point for lots of people we can still revert to more relaxed behavior.